salt/config/__init__.py

```diff
@@ -3649,6 +3653,21 @@ def get_id(opts, cache_minion_id=False):
     if opts.get('minion_id_lowercase'):
         newid = newid.lower()
         log.debug('Changed minion id %s to lowercase.', newid)
+
+    # Optionally remove a domain or many in a generated minion id
+    if opts.get('minion_id_remove_domain'):
+        rem_domain = opts.get('minion_id_remove_domain')
+        if isinstance(rem_domain, bool):
```

Comment:
```
if rem_domain is True:
```

Reply:
ok

Reply:
@markuskramerIgitt This is actually important for you to know in the future: you're only looking if `rem_domain` is a type of Boolean, but you do not know if it is True or False. Also, if you would just ask `if rem_domain:` then you get back a Boolean, and if it is True, still don't know what to do: remove specific domain, or any. Hence in these kind of situation we always explicitly looking for both type and value at the same time: `is None`, `is not None`, `is True` etc.

Hope it helps. :wink:

Reply:
It is the first time that I make use of `is True`, thank you for your explanation, @isbm.

I will "resolve" this conversation because GitHub sees open "requested changes".